src/js/components/JobForm.js

```diff
@@ -19,6 +19,14 @@ class JobForm extends SchemaForm {
 
   componentWillMount() {
     this.multipleDefinition = this.getNewDefinition();
+    // On edit hide the id field.
+    if (this.props.isEdit) {
+      this.multipleDefinition.general.definition.forEach(function (definition) {
+        if (definition.name === 'id') {
+          definition.formElementClass = 'hidden';
+        }
+      });
+    }
```

Comment:
i believe this piece of code belongs inside of `getNewDefinition`.

`JobForm` can implement it's own `getNewDefinition` by doing:

```js
getNewDefinition() {
  this.multipleDefinition = super.getNewDefinition();
  // your definition editing code here.
}
```